### DISCUSSION TITLE
What is the difference between @eclipse-glsp/dev and @eclipse-glsp/config

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1231] [Q&A] [answered]
When I try to learn how to setup my glsp project correctly I came across the GLSP modules `@eclipse-glsp/dev` and `@eclipse-glsp/config`. 
In some examples the module  `@eclipse-glsp/dev`  is referred in the `devDependencies` sometimes  `@eclipse-glsp/config`. 

What is the difference and should I add both modules to my GLSP project in general?

At least   `@eclipse-glsp/dev` seems to be important.

**Comment by tortmayr** [comment, marked as answer]
Hi @rsoika,

the purpose of the dev/config packages is explained in the corrsponding [README](https://github.com/eclipse-glsp/glsp/blob/master/dev-packages/README.md).

In a nutshell, these packags provide shared configuration for typescript, eslint, prettier, mocha etc.
It is split into three packages:
- `@eclipse-glsp/config`: Provides typescript, eslint and  prettier configuration
- `@eclipse-glsp/config-test`: Provides additional test configuration for mocha and the nyc test coverage tool
- `@eclipse-glsp/cli` A CLI tool with additional commands that are used in GLSP core projects. Most of these commands are probably not useful in adopter projects but in the end including it does not hurt as it's just a little package

In addition an all-in-one meta package (@eclipse-glsp/dev) is provided that includes all of the packages mentioned above.

What parts you want to include in your project is up to you.
You can either
- Simply consume the `@eclipse-glsp/dev` package and reuse the GLSP config. This is the most straight forward approach as you don't have to worry about configuration
- Only consume parts of the package e.g. just `@eclipse-glsp/config` if you don't use mocha/nyc in your project
- Don't use the GLSP config packages at all and manage the configuratio of typescript, eslint etc. on your own.

**Reply by rsoika** [reply]
Ok understand. 
So for me it seems that in the future I just need to add the following devDependency into my root `package.json` file to be happy 

```
  "devDependencies": {
    "@eclipse-glsp/dev": "~2.0.0",
    "inversify": "6.0.2",
    "lerna": "^6.6.2"
  }
```

**Reply by tortmayr** [reply]
Almost,
you also need to define the `typescript` version that you want to use. Anything `=4.9.x` should work, but I would recommend the latest version (5.x).

**Reply by rsoika** [reply]
ok,

```
  "devDependencies": {
    "@eclipse-glsp/dev": "~2.0.0",
    "inversify": "6.0.2",
    "lerna": "^6.6.2",
    "typescript": "^5.2.2"
  },
```

I also updated my [vscode-integration-guideline](https://github.com/imixs/open-bpmn-vscode-integration/blob/main/QUICKSTARTGUIDE.md) (still work in progress...)
